lib/python/rose/run_source_vc.py

```diff
@@ -22,6 +22,17 @@
 import os
 from rose.popen import RosePopener
 import sys
+import _io
+
+
+def _write_safely(msg, handle):
+    # Handle different combinations of handle and message types
+    if isinstance(handle, _io.BufferedWriter) and isinstance(msg, str):
+        handle.write(msg.encode('UTF-8'))
+    elif isinstance(handle, _io.StringIO) and isinstance(msg, bytes):
+        handle.write(msg.decode())
+    else:
+        handle.write(msg)
```

Comment:
Maybe we should have a `write_safely` function in a separate module - given that you have something with a similar name in a different module above?

Reply:
I'll add this as a separate PR. It's a bit much for now.